app/elements/pw-shell.html

```diff
@@ -340,15 +341,18 @@
         // Since the response is a document that's already parsed, don't
         // just innerHTML it (which converts it to a string and then reparses it),
         // and just use the nodes we need.
-        Polymer.dom(this).innerHTML = '';
-        var shell = doc.querySelector('pw-shell');
-
+        var docShell = doc.querySelector('pw-shell');
         var fragment = document.createDocumentFragment();
-        while (shell.firstChild) {
-          fragment.appendChild(shell.firstChild);
+        while (docShell.firstChild) {
+          fragment.appendChild(docShell.firstChild);
         }
 
-        Polymer.dom(this).appendChild(document.importNode(fragment, true));
+        // Replace shell contents.
+        var shell = Polymer.dom(this);
+        while (shell.firstChild) {
```

Comment:
Why is this better than `shell.innerHTML = '';`?

Reply:
DOM manipulations are faster than invoking the parser, I think (even for empty strings), though the real motivation behind this change was that setting `innerHTML` was causing issues.